src/erlfmt_cli.erl

```diff
@@ -246,48 +256,54 @@ read_stdin(Data) ->
+    parse_opts(Rest, Files, Exclude, Config#config{verbose = true});
+parse_opts([check | _Rest], _Files, _Exclude, #config{out = Out}) when Out =/= standard_out ->
     {error, "--check mode can't be combined write or replace mode"};
-parse_opts([check | Rest], Files, Config) ->
-    parse_opts(Rest, Files, Config#config{out = check});
-parse_opts([{print_width, Value} | Rest], Files, Config) ->
-    parse_opts(Rest, Files, Config#config{print_width = Value});
-parse_opts([require_pragma | _Rest], _Files, #config{pragma = insert}) ->
+parse_opts([check | Rest], Files, Exclude, Config) ->
+    parse_opts(Rest, Files, Exclude, Config#config{out = check});
+parse_opts([{print_width, Value} | Rest], Files, Exclude, Config) ->
+    parse_opts(Rest, Files, Exclude, Config#config{print_width = Value});
+parse_opts([require_pragma | _Rest], _Files, _Exclude, #config{pragma = insert}) ->
     {error, "Cannot use both --insert-pragma and --require-pragma options together."};
-parse_opts([require_pragma | Rest], Files, Config) ->
-    parse_opts(Rest, Files, Config#config{pragma = require});
-parse_opts([insert_pragma | _Rest], _Files, #config{pragma = require}) ->
+parse_opts([require_pragma | Rest], Files, Exclude, Config) ->
+    parse_opts(Rest, Files, Exclude, Config#config{pragma = require});
+parse_opts([insert_pragma | _Rest], _Files, _Exclude, #config{pragma = require}) ->
     {error, "Cannot use both --insert-pragma and --require-pragma options together."};
-parse_opts([insert_pragma | Rest], Files, Config) ->
-    parse_opts(Rest, Files, Config#config{pragma = insert});
-parse_opts([{files, NewFiles} | Rest], Files0, Config) ->
-    parse_opts(Rest, expand_files(NewFiles, Files0), Config);
-parse_opts([], [stdin], #config{out = Out}) when Out =/= standard_out, Out =/= check ->
+parse_opts([insert_pragma | Rest], Files, Exclude, Config) ->
+    parse_opts(Rest, Files, Exclude, Config#config{pragma = insert});
+parse_opts([{files, NewFiles} | Rest], Files, Exclude, Config) ->
+    parse_opts(Rest, expand_files(NewFiles, Files), Exclude, Config);
+parse_opts([{exclude_files, NewExcludes} | Rest], Files, Exclude, Config) ->
+    parse_opts(Rest, Files, expand_files(NewExcludes, Exclude), Config);
+parse_opts([], [stdin], _Exclude, #config{out = Out}) when Out =/= standard_out, Out =/= check ->
     {error, "stdin mode can't be combined with write options"};
-parse_opts([], [stdin], Config) ->
-    {format, [stdin], Config};
-parse_opts([], Files, Config) ->
+parse_opts([], [stdin], [], Config) ->
+    {format, [stdin], [], Config};
+parse_opts([], [stdin], [_Exclude | _], _Config) ->
+    {error, "stdin mode can't be combined with excluded files"};
+parse_opts([], Files, Exclude, Config) ->
     case lists:member(stdin, Files) of
         true -> {error, "stdin mode can't be combined with other files"};
-        false -> {format, lists:reverse(Files), Config}
+        false -> {format, lists:reverse(Files), lists:reverse(Exclude), Config}
```

Comment:
What if we checked files vs excludes in here and otherwise produced an `{error, ...}` tuple like we do for other cases? I think this could simplify `unprotected_with_config` that is already rather complex

Reply:
This was the first thing we considered, but what about specifying no files and only excluding files.
This should combine in a logical way with the default files we provide or those supplied in the rebar config.